src/routes/routes.ts

```diff
@@ -1,87 +1,95 @@
-const etapiSpecRoute = require('../etapi/spec');
-const etapiBackupRoute = require('../etapi/backup');
+import treeApiRoute = require('./api/tree');
+import notesApiRoute = require('./api/notes');
+import branchesApiRoute = require('./api/branches');
+import attachmentsApiRoute = require('./api/attachments');
+import autocompleteApiRoute = require('./api/autocomplete');
+import cloningApiRoute = require('./api/cloning');
+import revisionsApiRoute = require('./api/revisions');
+import recentChangesApiRoute = require('./api/recent_changes');
+import optionsApiRoute = require('./api/options');
+import passwordApiRoute = require('./api/password');
+import syncApiRoute = require('./api/sync');
+import loginApiRoute = require('./api/login');
+import recentNotesRoute = require('./api/recent_notes');
+import appInfoRoute = require('./api/app_info');
+import exportRoute = require('./api/export');
+import importRoute = require('./api/import');
+import setupApiRoute = require('./api/setup');
+import sqlRoute = require('./api/sql');
+import databaseRoute = require('./api/database');
+import imageRoute = require('./api/image');
+import attributesRoute = require('./api/attributes');
+import scriptRoute = require('./api/script');
+import senderRoute = require('./api/sender');
+import filesRoute = require('./api/files');
+import searchRoute = require('./api/search');
+import bulkActionRoute = require('./api/bulk_action');
+import specialNotesRoute = require('./api/special_notes');
+import noteMapRoute = require('./api/note_map');
+import clipperRoute = require('./api/clipper');
+import similarNotesRoute = require('./api/similar_notes');
+import keysRoute = require('./api/keys');
+import backendLogRoute = require('./api/backend_log');
+import statsRoute = require('./api/stats');
+import fontsRoute = require('./api/fonts');
+import etapiTokensApiRoutes = require('./api/etapi_tokens');
+import relationMapApiRoute = require('./api/relation-map');
+import otherRoute = require('./api/other');
+import shareRoutes = require('../share/routes');
+
+import etapiAuthRoutes = require('../etapi/auth');
+import etapiAppInfoRoutes = require('../etapi/app_info');
+import etapiAttachmentRoutes = require('../etapi/attachments');
+import etapiAttributeRoutes = require('../etapi/attributes');
+import etapiBranchRoutes = require('../etapi/branches');
+import etapiNoteRoutes = require('../etapi/notes');
+import etapiSpecialNoteRoutes = require('../etapi/special_notes');
+import etapiSpecRoute = require('../etapi/spec');
+import etapiBackupRoute = require('../etapi/backup');
+import { RequestHandlerParams } from 'express-serve-static-core';
+import { AppRequest, AppRequestHandler } from './route-interface';
 
 const csrfMiddleware = csurf({
     cookie: true,
     path: '' // empty, so cookie is valid only for the current path
-});
+    // TODO: Typescript complains that path does not exist in csurf options, but it's still in the
```

Comment:
I think you forgot to finish the comment :)
path is actually part of CookieOptions (see csurf types [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/csurf/index.d.ts) and the type of CookieOptions [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9236b5cd0c15232e8c0748363b46d720da570c05/types/express-serve-static-core/index.d.ts#L344)) so csurf is configured wrong here. We should anyways replace this soon, since it's deprecated.